_problems/fee-bumping.md

```diff
@@ -0,0 +1,96 @@
+Since miners prioritize transactions based on their feerate, the only effective way is to make sure participants can set the fee of time sensitive transactions and ideally even increase it repeatedly after the transaction has been broadcast if conditions change.
+
+There are two approaches available in Bitcoin today.
+
+1. *Replace-by-Fee* ([RBF]): The user simply signs a new transaction that is likely to replace the old one since it pays a higher fee.
+2. *Child-Pays-For-Parent* ([CPFP]): The original transaction remains unchanged but a new transaction with a higher fee is signed which spends from the old one increasing the average feerate of the pair.
+
+The goal of a researcher considering this problem should be to design a method for ensuring crucial layer-2 transactions are included in the chain in a timely fashion without all the complexity that the present state of affairs.
+Let's look at the implications of the current methods for layer-2 protocol designers.
+
+### Pre-signed Transactions
+
+Often the transaction an honest party is racing against the clock to confirm is a *pre-signed* transaction.
+It was signed when their counterparty was cooperative -- now the honest party needs to go on chain precisely because they are not cooperative.
+If fees become persistently higher than the pre-signed transaction's fees the party may miss their deadline.
+The obvious but suboptimal solution is to monitor fees closely and update any time-sensitive pre-signed transactions with the counterparty while they are online (this method is available via the [`update_fee`] message in lightning today).
+
+It is much more desirable to allow the party to spontaneously increase the fee.
+This can be done with [CPFP].
+For example, in the [Lightning] protocol commitment transactions are designed with [anchor outputs] so they can be easily fee bumped through CPFP.
+
+### Resource Usage of Replacement
+
+Allowing a mempool transaction to be replaced by higher fee paying transactions is that this may open up nodes to *denial of service* (DoS) attacks.
+Without replacement a malicious node's ability to get your node to process a transaction is limited by the number of UTXOs they own.
+With replacement an attacker can repeatedly spam your node with transactions that replace previous ones even if they only own a single UTXO.
+
+To prevent effective denial of service attacks, replacement transactions have to follow some rather strict rules as defined in [BIP125].
+In particular a replacement transaction must pay a higher absolute fee and a higher feerate than the transactions they are evicting and it must not evict more than 100 transactions.
+
+For layer-2 protocols this rules had the unintended consequence of introducing  "pinning" attacks (see [Related Research](#related-research)).
+This is usually done by adding a lot of descendants to a transaction that conflicts with the one the honest party needs confirm.
```

Comment:
Actually, the most straightforward pinning against a lot of L2 protocols is abuse of BIP125 rule 3 (absolute fee requirement). It's documented as scenario 2 on this branch : https://github.com/ariard/bitcoin/commit/67cbb701c108b77998f9bf036baabc188186e404 :)

Reply:
Hmm but is this still the case if CVE-2021-31876 is fixed? I changed the sentence to imply that adding descendants is just an example of how to do it and that the other rules can be abused too.

Reply:
Note that CVE-2021-31876 and BIP125 rule 3 are two differing pinning vectors. CVE-2021-31876 is quite easy to mitigate from a l2 protocol designer viewpoint by just signaling RBF on all in-protocol transactions.

Reply:
Understood but as I read it "scenario 2" is CVE-2021-31876 in the link provided? Did you mean scenario 1?

Reply:
Yes!